### PR TITLE
adds user creates a new folder

### PR DESCRIPTION
**Pull request by alfosco**
Adds functionality enabling a logged-in user to create a new folder.
-adds routes 'new' and 'create' for 'folders'
-adds 'new' and 'create' actions to 'folder' controller
-adds view (form) to create a new folder
-fixes error in test for 'update user' (we may want to add functionality for this to change the root folder name if a user changes their first name)

### CODE REVIEW

**Comment by JonathanSR** [comment]
@alfosco everything looks good.

**Comment by JonathanSR** [comment]
@alfosco everything looks good.

File: app/controllers/folders_controller.rb [hunk 1]
```diff
@@ -2,4 +2,25 @@ class FoldersController < ApplicationController
...
+      flash[:danger] = "You Suck"
+      render :new
+    end
+  end
+
+  private
+
+  def folder_params
+    params.require(:folder).permit(:name, :user_id, :parent_id)
+  end
```
Comment (cmacaulay) [inline app/controllers/folders_controller.rb]: 🎉

File: app/views/folders/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<h1>Create A New Folder</h1>
+
+<%= form_for @folder do |f| %>
+  <%= f.text_field :name, placeholder: "Folder Name" %>
+  <%= f.text_field :user_id, value: "#{current_user.id}", type: :hidden %>
+  <%= f.text_field :parent_id, value: "#{current_user.root_folder.id}", type: :hidden %>
+  <%= f.submit "Create Folder" %>
+<% end %>
```
Comment (cmacaulay) [inline app/views/folders/new.html.erb]: 🎉 great form!

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -1,9 +1,11 @@
...
 
+<%= link_to "Create New Folder", new_folder_path, class: "btn" %>
 <h3><%= @user.root_folder.name %></h3>
 
 <h3><% @user.folders.each do |folder| %></h3>
       <%= folder.name %>
+      <br>
     <% end %>
 <ul>
 <% @user.uploads.each do |upload| %>
```
Comment (cmacaulay) [inline app/views/users/show.html.erb]: thinking of probably making this a table, based off of the wireframes @alfosco @JonathanSR @AELSchauer ?

File: spec/features/user/user_can_update_account_spec.rb [hunk 4]
```diff
@@ -26,6 +26,8 @@
     end
 
     expect(page).to have_content("Purple")
-    expect(page).to_not have_content("Harry")
+    within("h1") do
+      expect(page).to_not have_content("Harry")
+    end
```
Comment (cmacaulay) [inline spec/features/user/user_can_update_account_spec.rb]: 👏 👏 
Great catch @alfosco

**Pull request merged**